### PR TITLE
Call DeleteTable from PgGateAPI

### PR DESCRIPTION
**Pull request by ahsank**
`DropDatabase` is already implemented https://github.com/futurewei-cloud/chogori-opengauss/blob/mainline/src/gausskernel/storage/access/k2/pg_gate_api.cpp#L195, Implementing `DropTable`

### CODE REVIEW

File: src/gausskernel/storage/access/k2/pg_gate_api.cpp [hunk 1]
```diff
@@ -371,12 +371,16 @@ K2PgStatus PgGate_NewDropTable(K2PgOid database_oid,
                             bool if_exist,
                             K2PgStatement *handle) {
   elog(LOG, "PgGateAPI: PgGate_NewDropTable %d, %d", database_oid, table_oid);
-  return K2PgStatus::NotSupported;
+  K2PgStatus status = pg_gate->GetCatalogClient()->DeleteTable(database_oid, table_oid);
```
Comment (johnfangAFW) [inline src/gausskernel/storage/access/k2/pg_gate_api.cpp:373]: The actual DropTable should be executed in PgGate_ExecDropTable(). Could you combine the following two methods,

K2PgStatus PgGate_NewDropTable(K2PgOid database_oid,
                            K2PgOid table_oid,
                            bool if_exist,
                            K2PgStatement *handle)
                            
and 

K2PgStatus PgGate_ExecDropTable(K2PgStatement handle)

into one method

K2PgStatus PgGate_ExecDropTable(K2PgOid database_oid,
                            K2PgOid table_oid,
                            bool if_exist)

File: src/gausskernel/storage/access/k2/pg_gate_api.cpp [hunk 2]
```diff
@@ -371,12 +371,16 @@ K2PgStatus PgGate_NewDropTable(K2PgOid database_oid,
...
+  if (if_exist && status.k2_code == 404) {
+      return K2PgStatus::OK;
+  }
+  return status;
 }
 
 K2PgStatus PgGate_ExecDropTable(K2PgStatement handle){
   elog(LOG, "PgGateAPI: PgGate_ExecDropTable");
-  return K2PgStatus::NotSupported;
+  return K2PgStatus::OK;
```
Comment (johnfangAFW) [inline src/gausskernel/storage/access/k2/pg_gate_api.cpp]: merge with the above method and modify the k2cat_cmds.cpp for this update.

**Review by johnfangAFW — CHANGES_REQUESTED** [review]

File: src/gausskernel/storage/access/k2/pg_gate_api.cpp (hunk 1 above)
Comment (ahsank) [reply src/gausskernel/storage/access/k2/pg_gate_api.cpp:373]: Done

File: src/gausskernel/storage/access/k2/pg_gate_api.cpp (hunk 2 above)
Comment (ahsank) [reply src/gausskernel/storage/access/k2/pg_gate_api.cpp]: Done

File: src/gausskernel/storage/access/k2/k2cat_cmds.cpp [hunk 3]
```diff
@@ -269,22 +269,12 @@ K2PgCreateTable(CreateStmt *stmt, char relkind, TupleDesc desc, Oid relationId,
...
-																									 false, /* if_exists */
-																									 &handle),
-																 &not_found);
-		const bool valid_handle = !not_found;
-		if (valid_handle)
-		{
-			HandleK2PgStatusIgnoreNotFound(PgGate_ExecDropTable(handle), &not_found);
-		}
-	}
+    bool not_found = false;
```
Comment (johnfangAFW) [inline src/gausskernel/storage/access/k2/k2cat_cmds.cpp]: not need to keep not_found here, remove it.

File: src/gausskernel/storage/access/k2/k2cat_cmds.cpp [hunk 4]
```diff
@@ -269,22 +269,12 @@ K2PgCreateTable(CreateStmt *stmt, char relkind, TupleDesc desc, Oid relationId,
...
+    HandleK2PgStatusIgnoreNotFound(PgGate_ExecDropTable(u_sess->proc_cxt.MyDatabaseId,
```
Comment (johnfangAFW) [inline src/gausskernel/storage/access/k2/k2cat_cmds.cpp]: remove not_found, just use HandleK2PgStatus() instead of HandleK2PgStatusIgnoreNotFound()

**Review by johnfangAFW — CHANGES_REQUESTED** [review]

File: src/gausskernel/storage/access/k2/k2cat_cmds.cpp (hunk 3 above)
Comment (ahsank) [reply src/gausskernel/storage/access/k2/k2cat_cmds.cpp]: Done

File: src/gausskernel/storage/access/k2/k2cat_cmds.cpp (hunk 4 above)
Comment (ahsank) [reply src/gausskernel/storage/access/k2/k2cat_cmds.cpp]: Done

**Review by johnfangAFW — APPROVED** [review]

**Pull request merged**